infra/modules/providers/azure/service-principal/main.tf

```diff
@@ -0,0 +1,25 @@
+module "azure-provider" {
+  source = "../provider"
+}
+
+locals {
+  rbac_true_false = "${var.create_for_rbac == "true" ? 1 : 0}"
+}
+
+data "azurerm_subscription" "sp" {}
+
+resource "azuread_application" "sp" {
+  count = "${local.rbac_true_false}"
```

Comment:
Curious: Why does Terraform allow a variation in the resource name? (azuread_application vs azurerm_azuread_application) - https://www.terraform.io/docs/providers/azurerm/r/azuread_application.html

Reply:
AzureAd and AzureRm are two different providers...